# tfs-ruby-4-0-x-rubygems-path-support.patch
#
# Target:  lib/rubygems/path_support.rb (pass1 (linux gnu/musl, macos))
# Applies: ruby 4.0.x (4.0.0 - 4.0.6)
# Source:  tebako gem lib/tebako/packager/rubygems_patch.rb RubygemsPatch#rubygems_path_support_patch
#
# Allow only packaged gems (from within the memfs): reset @home to
# Gem.default_dir and keep only @path entries under the memfs mount point
# (/__tebako_memfs__), unless TEBAKO_PASS_THROUGH is set.
diff --git a/lib/rubygems/path_support.rb b/lib/rubygems/path_support.rb
index 13091e2..a197b3b 100644
--- a/lib/rubygems/path_support.rb
+++ b/lib/rubygems/path_support.rb
@@ -25,7 +25,16 @@ class Gem::PathSupport
   #
   def initialize(env)
     @home = normalize_home_dir(env["GEM_HOME"] || Gem.default_dir)
+  # -- Start of tebako patch --
+    @home = Gem.default_dir unless @home.index("/__tebako_memfs__") == 0 unless env["TEBAKO_PASS_THROUGH"]
+# -- End of tebako patch --
     @path = split_gem_path env["GEM_PATH"], @home
+# -- Start of tebako patch --
+    @path.keep_if do |xpath|
+      xpath.index("/__tebako_memfs__") == 0
+    end unless env["TEBAKO_PASS_THROUGH"]
+# -- End of tebako patch --
+
 
     @spec_cache_dir = env["GEM_SPEC_CACHE"] || Gem.default_spec_cache_dir
   end
